data/hlint.yaml

```diff
@@ -379,6 +379,8 @@
     - warn: {lhs: mapM_ f (map g x), rhs: mapM_ (f . g) x}
     - warn: {lhs: mapM id, rhs: sequence}
     - warn: {lhs: mapM_ id, rhs: sequence_}
+    - hint: {lhs: "sequence [a]", rhs: "(:[]) <$> a"}
```

Comment:
Is the use of functor and `(:[])` really simpler? There seems a lot more "syntactic noise" in the RHS.

Reply:
It's true that it's noisy syntacticly, but it is conceptually simpler.

Imagine a case where someone had `mconcat <$> sequence [a, b]` and later edited it to `mconcat <$> sequence [a]`. After seeing the suggestion they may notice it should actually be just `a` now.

Reply:
The generalisation from `Applicative` to `Functor` is definitely nice and could also lead to wins elsewhere, but the RHS is definitely ugly. Maybe part of a new set of hints, for other uglifying generalisations?

Reply:
Maybe this is appropriate for the "generalise" group?